### ISSUE TITLE
Check failed: !HasOpenDirectories(). InitializeSandbox() called after unexpected directories have been opened. 

### ISSUE THREAD

**Issue by leonardochaia** [ianwalter/puppeteer#23]
Hi, I'm not sure if this is the place to ask this but perhaps you can redirect me to the correct place.
I have this repo https://github.com/leonardochaia/angular-simple-oidc where I'm using `ianwalter/puppeteer/` action to run Angular's tests on GitHub Actions.

It was working fine, but all of the sudden something changed either on the action side (which from what I can tell is not the case) or on the action's ubuntu environment. I'm really not sure.

Perhaps you have seen this before and may be able to point me in the right direction:
This is the relevant logs:
```
DevTools listening on ws://127.0.0.1:9222/devtools/browser/ea563c07-778a-48e4-ad4d-1dec04f1150f
[1014/145253.119752:FATAL:sandbox_linux.cc(382)] Check failed: !HasOpenDirectories(). InitializeSandbox() called after unexpected directories have been opened. This breaks the security of the setuid sandbox.
#0 0x5646c6b29879 base::debug::CollectStackTrace()
#1 0x5646c6a86c63 base::debug::StackTrace::StackTrace()
#2 0x5646c6a9b013 logging::LogMessage::~LogMessage()
#3 0x5646c81b954c service_manager::SandboxLinux::InitializeSandbox()
#4 0x5646ca6346ee content::(anonymous namespace)::ContentSandboxHelper::EnsureSandboxInitialized()
#5 0x5646c8077246 gpu::GpuInit::InitializeAndStartSandbox()
#6 0x5646ca633f10 content::GpuMain()
#7 0x5646c66cad94 content::ContentMainRunnerImpl::Run()
#8 0x5646c6716a8d service_manager::Main()
#9 0x5646c66c90b1 content::ContentMain()
#10 0x5646c671599d headless::(anonymous namespace)::RunContentMain()
#11 0x5646c671588f headless::RunChildProcessIfNeeded()
#12 0x5646c6714175 headless::HeadlessShellMain()
#13 0x5646c44a61b8 ChromeMain
#14 0x7f435f9982e1 <unknown>
#15 0x5646c44a602a _start
```

A [broken build](https://github.com/leonardochaia/angular-simple-oidc/commit/0b1a8cf5cae89706d7e9880f02ac2160043c4829/checks?check_suite_id=231365310)

A [running build](https://github.com/leonardochaia/angular-simple-oidc/commit/97c6a24b38705e3ee166339d8c7c78c5e3813955/checks?check_suite_id=231333471)

Any insight is welcome.
Thank you.

**Comment by ianwalter** [comment]
Checkout the options passed to puppeteer in my testing framework puppeteer plugin here and try to pass them in your karma setup: https://github.com/ianwalter/bff-puppeteer/blob/master/index.js#L17

**Comment by leonardochaia** [comment]
Hi Ian, thank you for your quick reply.
Unfortunately, that didn't work.
These are my [puppeteer params](https://github.com/leonardochaia/angular-simple-oidc/blob/6443bf30e4dbf4b3598b13756843566e5e49a5e1/projects/angular-simple-oidc/karma.conf.js#L36):

```typescript
['--no-sandbox', '--headless', '--disable-gpu', '--disable-translate', '--disable-extensions', '--disable-setuid-sandbox']
```

This is a [build with these params](https://github.com/leonardochaia/angular-simple-oidc/pull/36/checks?check_run_id=259369100)

**Comment by ianwalter** [comment]
https://github.com/ianwalter/bff-puppeteer/blob/master/index.js#L32

https://github.com/leonardochaia/angular-simple-oidc/blob/6443bf30e4dbf4b3598b13756843566e5e49a5e1/projects/angular-simple-oidc/karma.conf.js#L4

You need to tell puppeteer to use chrome in the image and not the chromium it downloads. You can use that env var to stop puppeteer from downloading chromium during yarn install as well.

**Comment by leonardochaia** [comment]
Hey Ian thank you so much for reviewing my setup.
I'm not sure how it was all working to be honest lol.
So I was using the puppeteer binary downloaded with the package.

I've decided to add the env vars to my workflow config instead of them being in the code.

For future references, [change I did to fix this](https://github.com/leonardochaia/angular-simple-oidc/pull/36/files)

Thank you so much!
